pkg/controller/utils/utils.go

```diff
@@ -299,3 +300,23 @@ func GetExpectedTyphaScale(nodes int) int {
 	}
 	return typhas
 }
+
+// GetElasticLicenseType returns the license type from elastic-licensing ConfigMap that ECK operator keeps updated.
+func GetElasticLicenseType(ctx context.Context, cli client.Client, logger logr.Logger) (render.ElasticLicenseType, error) {
+	cm := &corev1.ConfigMap{}
+	err := cli.Get(ctx, client.ObjectKey{Name: render.ECKLicenseConfigMapName, Namespace: render.ECKOperatorNamespace}, cm)
+	if err != nil {
+		// If ConfigMap is not available, it means ECK operator is not running yet, so log the information and proceed
+		if errors.IsNotFound(err) {
+			logger.Info("%s ConfigMap not found yet", render.ECKLicenseConfigMapName)
+			return render.ElasticLicenseTypeUnknown, nil
+		}
+		return render.ElasticLicenseTypeUnknown, err
+	}
+	license, ok := cm.Data["eck_license_level"]
+	if !ok {
+		return render.ElasticLicenseTypeUnknown, fmt.Errorf("eck_license_level not available.")
+	}
+
+	return render.ElasticLicenseType(license), nil
```

Comment:
I think that you should write a function `render.StrToElasticLicenseType` to guarantee this only returns types that we want to consider. For instance, if don't want to consider somebody who put in a platinum license then this would return unknown if the type was platinum, then we just check if the type is unknown. 

This gives the Unknown type a real meaning besides just what to return for an error, it's Unknown to us but not necessarily an error, but a caller _could_ treat it as an error if it requires a known license type.

There's almost no point in using a specific type if you're just going to cast every string value as the ElasticLicenseType since it doesn't validate that it's a type you've defined.